packages/components/src/unit-control/index.native.js

```diff
@@ -20,125 +20,133 @@ import { CSS_UNITS } from './utils';
-	...props
-} ) {
-	const pickerRef = useRef();
-	const anchorNodeRef = useRef();
+const UnitControl = memo(
+	( {
+		currentInput,
+		label,
+		value,
+		onChange,
+		onUnitChange,
+		initialPosition,
+		min,
+		max,
+		separatorType,
+		units = CSS_UNITS,
+		unit,
+		getStylesFromColorScheme,
+		decimalNum,
+		...props
+	} ) => {
+		const pickerRef = useRef();
+		const anchorNodeRef = useRef();
 
-	function onPickerPresent() {
-		if ( pickerRef?.current ) {
-			pickerRef.current.presentPicker();
+		function onPickerPresent() {
+			if ( pickerRef?.current ) {
+				pickerRef.current.presentPicker();
+			}
 		}
-	}
 
-	const currentInputValue = currentInput === null ? value : currentInput;
-	const initialControlValue = isFinite( currentInputValue )
-		? currentInputValue
-		: initialPosition;
+		const currentInputValue = currentInput === null ? value : currentInput;
+		const initialControlValue = isFinite( currentInputValue )
+			? currentInputValue
+			: initialPosition;
+
+		const unitButtonTextStyle = getStylesFromColorScheme(
+			styles.unitButtonText,
+			styles.unitButtonTextDark
+		);
 
-	const unitButtonTextStyle = getStylesFromColorScheme(
-		styles.unitButtonText,
-		styles.unitButtonTextDark
-	);
+		/* translators: accessibility text. Inform about current unit value. %s: Current unit value. */
+		const accessibilityLabel = sprintf( __( 'Current unit is %s' ), unit );
 
-	const renderUnitButton = () => {
 		const accessibilityHint =
 			Platform.OS === 'ios'
 				? __( 'Double tap to open Action Sheet with available options' )
 				: __(
 						'Double tap to open Bottom Sheet with available options'
 				  );
 
-		/* translators: accessibility text. Inform about current unit value. %s: Current unit value. */
-		const accessibilityLabel = sprintf( __( 'Current unit is %s' ), unit );
+		const renderUnitButton = useMemo( () => {
+			return (
+				<TouchableWithoutFeedback
+					onPress={ onPickerPresent }
+					accessibilityLabel={ accessibilityLabel }
+					accessibilityRole="button"
+					accessibilityHint={ accessibilityHint }
+				>
+					<View style={ styles.unitButton }>
+						<Text style={ unitButtonTextStyle }>{ unit }</Text>
+					</View>
+				</TouchableWithoutFeedback>
+			);
+		}, [
+			onPickerPresent,
+			accessibilityLabel,
+			accessibilityHint,
+			unitButtonTextStyle,
+			unit,
+		] );
 
-		return (
-			<TouchableWithoutFeedback
-				onPress={ onPickerPresent }
-				accessibilityLabel={ accessibilityLabel }
-				accessibilityRole="button"
-				accessibilityHint={ accessibilityHint }
-			>
-				<View style={ styles.unitButton }>
-					<Text style={ unitButtonTextStyle }>{ unit }</Text>
-				</View>
-			</TouchableWithoutFeedback>
-		);
-	};
+		const getAnchor = () =>
+			anchorNodeRef?.current
+				? findNodeHandle( anchorNodeRef?.current )
+				: undefined;
 
-	const getAnchor = () =>
-		anchorNodeRef?.current
-			? findNodeHandle( anchorNodeRef?.current )
-			: undefined;
+		const renderUnitPicker = useCallback( () => {
```

Comment:
The same situation is here. The `getAnchor` is created every time and if it is set as a dependency of `useCallback` then this `useCallback` is called per each render as well (could be removed and it will work the same)